index.js

```diff
@@ -321,6 +326,38 @@ const captureWebsite = async (input, options) => {
 		await options.beforeScreenshot(page, browser);
 	}
 
+	if (screenshotOptions.fullPage) {
+		// Get the height of the rendered page
+		const bodyHandle = await page.$('body');
+		const ppHeight = await bodyHandle.boundingBox();
+		await bodyHandle.dispose();
+
+		// Scroll one viewport at a time, pausing to let content load
+		const viewportHeight = viewportOptions.height;
+		let viewportIncr = 0;
+		while (viewportIncr + viewportHeight < ppHeight) {
+			/* eslint-disable no-await-in-loop */
+			await page.evaluate(_viewportHeight => {
+				/* eslint-disable no-undef */
+				window.scrollBy(0, _viewportHeight);
+				/* eslint-enable no-undef */
+			}, viewportHeight);
+			await sleep(20);
+			/* eslint-enable no-await-in-loop */
+			viewportIncr += viewportHeight;
+		}
+
+		// Scroll back to top
+		await page.evaluate(_ => {
+			/* eslint-disable no-undef */
+			window.scrollTo(0, 0);
+			/* eslint-enable no-undef */
+		});
+
+		// Some extra delay to let images load
+		await page.waitForFunction(imagesHaveLoaded, {timeout: 60});
```

Comment:
I don't think the user would expect it to take up to one minute. If we keep this, it needs to be documented.

Reply:
Might also be added as an argument with default value set to timeout 60. Might want to comment on this or implement some other better functionality? Maybe it's also worth looking into alternate functions, such as the `waitForNavigation` Vasile mentioned earlier.

> This
> 
> ```js
> await wait(100);
> ```
> 
> is not good.
> 
> You should wait actually for images and other snippets to load, not giving them 100ms...
> The waitForNavigation method waiting for a network idle should be used instead.

In case it might be useful for documentation purposes; I retrieved part of the solution from https://stackoverflow.com/questions/51651830/how-to-wait-for-all-images-to-load-from-page-evaluate-function-in-puppeteer-when

Reply:
I would look into something better.

Reply:
using `page.waitForNavigation` as per pointed by https://github.com/sindresorhus/capture-website/pull/40#pullrequestreview-390231349